### ISSUE TITLE
LB port 443 does not respect ssl checkbox.

### ISSUE THREAD

**Issue by kaos** [rancher/rancher#5621]
**Rancher Version:**
1.1.2

**Docker Version:**

**OS and where are the hosts located? (cloud, bare metal, etc):**

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
cattle

**Steps to Reproduce:**
Create LB service, listening on port 443, http mode, NOT ssl.

**Results:** 
LB service, listening on port 443, http mode, **SSL**.

**Expected:**
I can see what's going on, but in my case, I explicitly do not want ssl although it is using port 443.

The reason is that it is for internal traffic between LB services, but the "inner most" services doesn't know there is another LB service in front of them, and they expect that SSL be used on the front towards the clients, so they expect to use port 443, but I don't want the overhead of re-encrypting the inter-LB traffic.

**Comment by vincent99** [comment]
I haven't actually tested this yet, but I'm not sure how it's possible; You presumably didn't specify a SSL cert (since the UI won't let you without SSL checked...), so what certificate is being used to terminate the connection?

It's possible there's some bug where we or haproxy assume if you're doing 443+HTTP then you want SSL termination, because 443 + HTTP + [not SSL termination] really makes no sense.  The balancer can't look at the request's HTTP headers/do host-based routing/inject X-Forwarded-For/etc unless it's a plaintext HTTP request or it is terminating the TLS connection.

So you really want 443 + TCP + [not SSL] to pass the encrypted request through to the target service unchanged.

**Comment by kaos** [comment]
Oh, sorry, that was missing from the steps to reproduce. I've "re-used" this LB from the days when it did terminate the ssl, so there are certs added already, and there used to be a SSL enabled port 443. But I have since cloned it and changed the ports around, and when I finally went back to using port 443 again (but attempting to not check ssl) it wouldn't work. So perhaps this is rather abuse of the settings, and ought to deny my attempt to create such a service in the first place :) Wouldn't have been the first time I try something that [doesn't work](https://forums.rancher.com/t/lb-targeting-alias-service/803), right :P 

But I don't want to pass the request encrypted to the target service, the TLS connection is terminated at the first HAproxy, then I simply want to pass the request downstream unencrypted (as it is going on the rancher network, it deem it safe enough).

I'll retry my operation on a fresh LB service to see how it behaves.. also, for now I've resolved to re-encrypt the traffic between the proxies, so it is no big issue atm. ;)

**Comment by kaos** [comment]
Ok, this was easy to reproduce and test, and think some minor checks/tweaks in the dialog will be sufficient.

I was unable to get rid of the SSL status on a port when cloning. But, it seems to work any way if I deselect the SSL certificate. So I end up with a LB that says it does SSL, but has no cert, and the config for that port does not do SSL.

```
frontend ab4cf3cc-79ab-45ff-9a8b-f5e5042dfc78_443_frontend
        bind *:443
        mode http
```

![image](https://cloud.githubusercontent.com/assets/72965/17393799/30c2144a-5a26-11e6-9d6f-8d144b864bc4.png)

So, I think that there are two things to fix,

1) When cloning from a LB service, and you deselect all SSL ports, clear all SSL certificate configs upon save.

2) .. I had two things in mind, but now it fell out of my head and I can't find it again.. 😕

**Comment by tfiduccia** [comment]
@kaos - That is the edit screen not a clone.  If you go to the menu and click on Clone, the Ports are actually deselectable
![image](https://cloud.githubusercontent.com/assets/11514927/17419134/e319ab10-5a50-11e6-82e7-666c056e9732.png)

**Comment by kaos** [comment]
@tfiduccia I know, sorry if I was unclear, skipped ahead to the end result. In the clone menu, I deslected the ssl checkbox, as described in text, the screen shot was only to show what the cloned service looks like in the edit view, that the ssl option was still ticked.

If you do this, you'll get the same:

1) Create LB service, 443 SSL
2) Clone to new service, 443 No SSL, delete all certificate options
3) Edit Cloned service, you'll see that port 443 still has ssl enabled in the box, but if you inspect the haproxy config, it doesn't enable ssl in the frontend config for port 443. Thus, the UI/API info contradict the actual generated haproxy config.

**Comment by vincent99** [comment]
This shouldn't be relevant in 1.2, ssl-ness is now part of the protocol choice for a rule.

**Comment by tfiduccia** [comment]
@kaos 
This is resolved in 1.2
Version - master 11/06
Verified fixed

**Comment by kaos** [comment]
Thanks! :)
